pages/en/contrib/style-guide-es6.md

```diff
@@ -0,0 +1,764 @@
+old `require('util').inherits` approach.
+
+**Good**:
+
+```js
+class MyConnector extends BaseConnector {
+  constructor(settings, dataSource) {
+    // ...
+  }
+
+  set(modelName, key, value, options, callback) {
+    // ...
+   }
+}
+```
+
+**Bad**:
+
+```js
+function MyConnector(settings, dataSource) {
+  // ...
+}
+util.inherits(MyConnector, BaseConnector);
+
+MyConnector.prototype.set = function(modelName, key, value, options, callback) {
+  // ...
+};
+```
+
+### One argument per line
+
+Once you cannot fit all arguments into a single line shorter than 80 characterss, it's better to place each argument on a new line.
+
+**Good**:
+
+```js
+TestModel.find(
+  {where: {id: '1'}},
+  {notify: false},
+  function(err, list) {
+    ...
+  });
+```
+
+**Bad**:
+
+```js
+TestModel.find(
+  {where: {id: '1'}}, {notify: false},
+  function(err, list) {
+    ...
+  });
+```
+
+**Bad**:
+
+```js
+TestModel.find({where: {id: '1'}},
+  {notify: false},
+  function(err, list) {
+    ...
+  });
+```
+
+{% include note.html title="EXCEPTION:" content="When the callback function is the only argument that overflows character limit, you can put only this argument on a new line.
+" %}
+
+For example:
+
+```js
+TestModel.find({where: {id: '1'}},
+  function(err, list) {
+    ...
+  });
+```
+
+{% include note.html title="EXCEPTION:" content="When fixing existing code, it's better to preserve indentation of the inner function body and do not indent the second line.
+" %}
+
+For example:
+
+```js
+TestModel.find({where: {id: '1'}},
+function(err, list) {
+  ...
+});
+```
+
+{% include note.html title="EXCEPTION:" content="When the arguments are only short primitive values (strings, numbers) or short variable/property references, you can collapse them on the same line.
+" %}
+
+**Good**:
+
+```js
+console.error('Unhandled array of errors for request %s %s\n',
+  req.method, req.url, errors);
+
+console.error(
+  'Unhandled array of errors for request %s %s\n',
+  req.method, req.url, errors);
+```
+
+**Bad**:
+
+```js
+console.error(
+  'Unhandled array of errors for request %s %s\n',
+  req.method,
+  req.url,
+  errors);
+```
+
+### Indentation of multi-line expressions in `return`
+
+Indent the second and all next lines by one level.
+
+**Good**:
+
+```js
+return (testInEquality({gte: example.between[0]}, value) &&
+  testInEquality({lte: example.between[1]}, value) &&
+  testInEquality({lte: example.between[2]}, value));
+```
+
+**Bad**:
+
+```js
+return (testInEquality({gte: example.between[0]}, value) &&
+       testInEquality({lte: example.between[1]}, value) &&
+       testInEquality({lte: example.between[2]}, value));
+```
+
+### Indentation of multi-line expressions in `if`
+
+Prefer to extract the multi-line expression to a variable, as it is easiest to read. Use a good variable name to describe the condition you are building.
+
+When not feasible, then indent the second and next lines by two levels.
+
+**Good**:
+
+```
+TBD
```

Comment:
I guess you meant to put some content in here?

Reply:
I'll remove this section in favour of "Still acceptable" below.